fineract-core/src/main/java/org/apache/fineract/infrastructure/codes/domain/CodeValueRepository.java

```diff
@@ -18,12 +18,39 @@
  */
 package org.apache.fineract.infrastructure.codes.domain;
 
+import java.util.Optional;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 public interface CodeValueRepository extends JpaRepository<CodeValue, Long>, JpaSpecificationExecutor<CodeValue> {
 
     CodeValue findByCodeNameAndId(String codeName, Long id);
 
     CodeValue findByCodeNameAndLabel(String codeName, String label);
+
+    @Query(value = """
+            SELECT
+               mv2.code_score AS ciudad_code_score,
+               mv2.code_value AS ciudad_code_value,
+               mcv.code_score AS departamento_code_score,
+               mcv.code_id AS departamento_code_id,
+               mcv.code_value AS departamento_code_value
+            FROM m_code_value mcv
+            INNER JOIN (
+                SELECT *,
+                CASE
+                    WHEN LEFT(LPAD(code_score::TEXT, 5, '0'), 2)::INTEGER < 10 THEN
+                        LEFT(LPAD(code_score::TEXT, 5, '0'), 2)::INTEGER::VARCHAR
+                    ELSE
+                        LEFT(LPAD(code_score::TEXT, 5, '0'), 2)
+                END AS ciudadCoderScore
+                FROM m_code_value mcv2
+                WHERE code_id = 41
+            ) mv2
+            ON mv2.ciudadCoderScore = mcv.code_score
+            WHERE mcv.code_id = 40 AND mv2.code_score = :codeScore
```

Comment:
How reliable is this? Using a literal value for the `code_id`. Maybe it's better to use the code name instead. Something like this:
`WHERE mcv.code_id = (SELECT id FROM m_code WHERE name = 'XXX')

Reply:
i already update